packages/network-enablement-controller/src/NetworkEnablementController.ts

```diff
@@ -247,6 +247,20 @@ export class NetworkEnablementController extends BaseController<
     });
   }
 
+  /**
+   * Checks if a network is enabled.
+   *
+   * @param chainId - The chain ID of the network to check. Can be either:
+   * - A Hex string (e.g., '0x1' for Ethereum mainnet) for EVM networks
+   * - A CAIP-2 chain ID (e.g., 'eip155:1' for Ethereum mainnet, 'solana:mainnet' for Solana)
+   * @returns True if the network is enabled, false otherwise
+   */
+  isNetworkEnabled(chainId: Hex | CaipChainId): boolean {
+    const derivedKeys = deriveKeys(chainId);
+    const { namespace, storageKey } = derivedKeys;
+    return this.state.enabledNetworkMap[namespace]?.[storageKey] ?? false;
+  }
```

Comment:
Nice!
Technically this doesn't do any state manipulation stuff, could be a util or selector?

Reply:
Not a blocker, we can refactor out if this controller gets really large.